keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/credential/Ssha256PasswordHashProvider.java

```diff
@@ -0,0 +1,86 @@
+package io.cloudtrust.keycloak.credential;
+
+import org.keycloak.common.util.Base64;
+import org.keycloak.credential.hash.PasswordHashProvider;
+import org.keycloak.models.PasswordPolicy;
+import org.keycloak.models.credential.PasswordCredentialModel;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+/**
+ * Implementation of the Salted SHA256 password hash algorithm.
+ */
+public class Ssha256PasswordHashProvider implements PasswordHashProvider {
+
+    @Override
+    public boolean policyCheck(PasswordPolicy policy, PasswordCredentialModel credential) {
+        return policy != null && credential.getPasswordCredentialData().getAlgorithm().equals(Ssha256PasswordHashProviderFactory.ID);
```

Comment:
Nothing to check against policy value?
Example: Shouldn't policy.getHashAlgorithm() be Ssha256PasswordHashProviderFactory.ID too?

Reply:
I don't think we need to do it. From my understanding, if the hashing algorithm is not SSHA256 this function will never be called.
In Keycloak's implementation of Argon2 and Pbkdf2 they aren't checking this and I guess it would be redundant to do it